### PR TITLE
Added documentation to Brainduino

### PR DESCRIPTION
**Pull request by rafaelcn**
Brainduino is now almost fully documented. Please, make a full review of it @luizperes. `BrainDelegate` class still lacks of docs though, if approved I will generate the html docs.

Signed-off-by: Rafael Campos Nunes rafaelnunes@engineer.com

### CODE REVIEW

**Comment by rafaelcn** [comment]
Weird indentation :thought_balloon:

File: Brainduino/Brain.cpp [hunk 1]
```diff
@@ -7,28 +7,28 @@
...
 }
 
-void Brain::set(Print *printer, Stream *streamIn, BrainDelegate *delegate,
+void Brain::set(Print *printer, Stream *stream_in, BrainDelegate *delegate,
                 char const *code)
 {
     reset();
     _printer = printer;
-    _streamIn = streamIn;
+    _streamIn = stream_in;
```
Comment (luizperes) [inline Brainduino/Brain.cpp:26]: I imagine that you will have a problem in this line. Could you rename the variable _streamIn?

File: Brainduino/Brain.cpp [hunk 2]
```diff
@@ -42,24 +42,24 @@ void Brain::reset()
...
 
 int Brain::exec()
 {
     int _index = 0;
     int _action = 0;
     int _jumps[STACK_SIZE];
-    int _indexJumps = 0;
+    int _index_jumps = 0;
     int _iterations[STACK_SIZE];
     int _indexIterations = 0;
```
Comment (luizperes) [inline Brainduino/Brain.cpp:64]: Could you rename the variable _indexIteration as well to _index_iterations?

**Comment by luizperes** [comment]
Hi @rafaelcn, thanks for your PR. Could you check my considerations?

**Comment by ryukinix** [comment]
@luizperes Nice comment.

File: Brainduino/Brain.cpp (hunk 2 above)
Comment (rafaelcn) [reply Brainduino/Brain.cpp:64]: Slip of the eye, thanks for the notes.

File: Brainduino/Brain.cpp (hunk 1 above)
Comment (rafaelcn) [reply Brainduino/Brain.cpp:26]: Yep! Thanks for the note.

**Comment by luizperes** [comment]
LGTM !

**Review by luizperes — APPROVED** [review]

**Comment by rafaelcn** [comment]
What have you done? I didn't send you commits fixing those notes .-.

**Comment by luizperes** [comment]
I saw it. I merged and fixed it myself because I needed to update the [Brain Station](https://github.com/brain-labs/brainstation). It is ok at this version https://github.com/brain-labs/brainduino/commit/f7fd500913e49d77683cba158e2d74cbd4520c2a. Sorry for the trouble @rafaelcn

**Comment by rafaelcn** [comment]
No problem then :-)

**Pull request merged**